### PR TITLE
Api600 logical interconnects

### PR DESCRIPTION
**Pull request by madhav-bharadwaj**
### Description
Support logical Interconnects for API600

### Check List
- [X] New functionality includes testing.
  - [X] All tests pass (`$ rake test`).
- [X] New functionality has been documented in the README if applicable.
  - [X] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [X] New endpoints supported are updated in the endpoints-support.md file.
- [X] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: endpoints-support.md [hunk 1]
```diff
@@ -249,34 +249,34 @@ OneviewSDK::Datacenter.find_by(@client, width: 11000).map(&:remove)
...
-|<sub>/rest/logical-interconnects/{id}/ethernetSettings</sub>                             | PUT      | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
-|<sub>/rest/logical-interconnects/{id}/firmware</sub>                                     | GET      | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
-|<sub>/rest/logical-interconnects/{id}/firmware</sub>                                     | PUT      | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
-|<sub>/rest/logical-interconnects/{id}/forwarding-information-base</sub>                  | GET      | :heavy_multiplication_x:       | :heavy_multiplication_x:       | :heavy_multiplication_x:       |
-|<sub>/rest/logical-interconnects/{id}/forwarding-information-base</sub>                  | POST     |  :heavy_multiplication_x:      | :heavy_multiplication_x:       | :heavy_multiplication_x:       |
+|<sub>/rest/logical-interconnects</sub>                                                   | GET      | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |  :white_check_mark:   |
```
Comment (soodpr) [inline endpoints-support.md]: all endpoints are not marked for the resource. Please check the UI view

File: examples/shared_samples/logical_interconnect.rb [hunk 2]
```diff
@@ -56,7 +58,8 @@
   name:  'li_et01',
   smartLink:  false,
   privateNetwork:  false,
-  connectionTemplateUri: nil
+  connectionTemplateUri: nil,
+  type: 'ethernet-networkV4'
```
Comment (soodpr) [inline examples/shared_samples/logical_interconnect.rb]: we don't need type here as we are setting default values in lib.

File: examples/shared_samples/logical_interconnect.rb [hunk 3]
```diff
@@ -266,7 +270,7 @@
 puts "\nOperations with scopes"
 begin
   # Scope class used in this sample
-  scope_class = OneviewSDK.resource_named('Scope', @client.api_version) unless @client.api_version.to_i <= 200
+  scope_class = OneviewSDK.resource_named('Scope', @client.api_version) unless @client.api_version.to_i <= 200 || client.api_version.to_i >= 600
```
Comment (soodpr) [inline examples/shared_samples/logical_interconnect.rb]: we have to modify this condition once we support 800 API. Its better to give if condition here

File: spec/integration/resource/api600/synergy/logical_interconnect/update_spec.rb [hunk 4]
```diff
@@ -1,22 +1,22 @@
...
+  let(:ethernet_class) { OneviewSDK::API600::Synergy::EthernetNetwork }
+  let(:interconnect_class) { OneviewSDK::API600::Synergy::Interconnect }
   let(:log_int_name) { LOG_INT2_NAME }
   let(:encl_name) { ENCL2_NAME }
 
-  it_behaves_like 'LogicalInterconnectUpdateExample', 'integration api500 context'
+  it_behaves_like 'LogicalInterconnectUpdateExample', 'integration api600 context'
 
-  it_behaves_like 'ScopeHelperMethodsExample', OneviewSDK::API500::Synergy::Scope do
+  it_behaves_like 'ScopeHelperMethodsExample', OneviewSDK::API600::Synergy::Scope do
```
Comment (soodpr) [inline spec/integration/resource/api600/synergy/logical_interconnect/update_spec.rb]: we don't scope here as we have removed PATCH

File: examples/shared_samples/logical_interconnect.rb (hunk 3 above)
Comment (madhav-bharadwaj) [reply examples/shared_samples/logical_interconnect.rb]: As Discussed, Keeping as is.

File: examples/shared_samples/logical_interconnect.rb (hunk 2 above)
Comment (madhav-bharadwaj) [reply examples/shared_samples/logical_interconnect.rb]: I'll change this. Forgot to delete after running the example.

**Pull request merged**